core/domain/services/assets/CoreAssetManager.php

```diff
@@ -0,0 +1,357 @@
+            EE_GLOBAL_ASSETS_URL . 'scripts/jquery.validate.additional-methods.min.js',
+            array(CoreAssetManager::JS_HANDLE_JQUERY_VALIDATE)
+        )
+        ->setVersion('1.15.0');
+    }
+
+
+    /**
+     * accounting.js for performing client-side calculations
+     *
+     * @since $VID:$
+     * @throws DuplicateCollectionIdentifierException
+     * @throws InvalidDataTypeException
+     * @throws InvalidEntityException
+     */
+    private function loadAccountingJs()
+    {
+        //accounting.js library
+        // @link http://josscrowcroft.github.io/accounting.js/
+        $this->addJavascript(
+            CoreAssetManager::JS_HANDLE_ACCOUNTING_CORE,
+            EE_THIRD_PARTY_URL . 'accounting/accounting.js',
+            array(CoreAssetManager::JS_HANDLE_UNDERSCORE)
+        )
+        ->setVersion('0.3.2');
+
+        $currency_config = $this->currency_config;
+        $this->addJavascript(
+            CoreAssetManager::JS_HANDLE_EE_ACCOUNTING,
+            EE_GLOBAL_ASSETS_URL . 'scripts/ee-accounting-config.js',
+            array(CoreAssetManager::JS_HANDLE_ACCOUNTING_CORE)
+        )
+        ->setLocalizationCallback(
+            function () use ($currency_config) {
+                 wp_localize_script(
+                     CoreAssetManager::JS_HANDLE_EE_ACCOUNTING,
+                     'EE_ACCOUNTING_CFG',
+                     array(
+                         'currency' => array(
+                             'symbol'    => $currency_config->sign,
+                             'format'    => array(
+                                 'pos'  => $currency_config->sign_b4 ? '%s%v' : '%v%s',
+                                 'neg'  => $currency_config->sign_b4 ? '- %s%v' : '- %v%s',
+                                 'zero' => $currency_config->sign_b4 ? '%s--' : '--%s',
+                             ),
+                             'decimal'   => $currency_config->dec_mrk,
+                             'thousand'  => $currency_config->thsnds,
+                             'precision' => $currency_config->dec_plc,
+                         ),
+                         'number'   => array(
+                             'precision' => $currency_config->dec_plc,
+                             'thousand'  => $currency_config->thsnds,
+                             'decimal'   => $currency_config->dec_mrk,
+                         ),
+                     )
+                 );
+            }
+        )
+        ->setVersion();
+    }
+
+
+    /**
+     * registers assets for cleaning your ears
+     *
+     * @param JavascriptAsset $script
+     */
+    public function loadQtipJs(JavascriptAsset $script)
+    {
+        // qtip is turned OFF by default, but prior to the wp_enqueue_scripts hook,
+        // can be turned back on again via: add_filter('FHEE_load_qtip', '__return_true' );
+        if (
+            $script->handle() === CoreAssetManager::JS_HANDLE_EE_WP_PLUGINS_PAGE
```

Comment:
This is a new condition that I don't recognize from before the changes.  Prior to your changes there was only the `apply_filters`.  The qtip js is something the _entire_ ee admin references, not just the page the script with the handle `self::JS_HANDLE_EE_WP_PLUGINS_PAGE` loads on.

This may have been something you did by mistake, but I wonder if in the future when you make logic changes like this in refactoring code if it would be helpful if you document for the reviewer why the change was made?

Reply:
sure I can explain this change in a phpdoc comment.
This may not be necessary at all, but the reason for it was to ensure that the qTip assets loaded at the same time as they did in the old code.

Previously the code was like this:

```php
        $this->loadAccountingJs(); 
        $this->loadQtipJs(); 
        $this->registerAdminAssets(); 
```

but the qTip code could not be converted to use the new asset manager in this branch because that `loadQtipJs()` method was just doing this:

```php
    private function loadQtipJs() 
    { 
        // qtip is turned OFF by default, but prior to the wp_enqueue_scripts hook, 
        // can be turned back on again via: add_filter('FHEE_load_qtip', '__return_true' ); 
        if (apply_filters('FHEE_load_qtip', false)) { 
            EEH_Qtip_Loader::instance()->register_and_enqueue(); 
        } 
    } 
```

So without getting into completely converting the logic in `EEH_Qtip_Loader` (which is fairly complex) we needed to run that same logic before the logic that was previously in `registerAdminAssets()`. That logic was simply registering assets under the 'ee-wp-plugins-page' handle. I did this using the following:

```php
        $this->loadAccountingJs();
        add_action(
            'AHEE__EventEspresso_core_services_assets_Registry__registerScripts__before_script',
            array($this, 'loadQtipJs')
        );
        $this->registerAdminAssets();
```

which hooks the `loadQtipJs()` method into the block of code in `Registry::registerScriptsAndStyles()` (need to update the above hook name):

```php
            do_action(
                'AHEE__EventEspresso_core_services_assets_Registry__registerScriptsAndStyles__before_script',
                $script
            );
            wp_register_script(
                $script->handle(),
                $script->source(),
                $script->dependencies(),
                $script->version(),
                $script->loadInFooter()
            );
```

That hook passes the current script object along as a parameter, so if we wanted to do something immediately before registering the script using the 'ee-wp-plugins-page' handle, then we need to hook into 'AHEE__EventEspresso_core_services_assets_Registry__registerScripts__before_script' with a cllback that checks for that script handle:


```php
    public function callback(JavascriptAsset $script)
    {
        if (
            $script->handle() === CoreAssetManager::JS_HANDLE_EE_WP_PLUGINS_PAGE
        ) {
            // do something
        }
    }
```

IF it doesn't matter that the qTip assets are registered before the 'ee-wp-plugins-page' script, then I can maybe add another hookpoint to  `Registry::registerScriptsAndStyles()` after all of the other scripts run, and use that to trigger the `loadQtipJs()` method, but then that would mean the qTip stuff is happening after all other scripts load. The way I did it now ensures it happens at the exact same time it used to.

Reply:
> sure I can explain this change in a phpdoc comment.

Just to be clear,  I was not suggesting all changes need to be made in a phpdoc comment.  I was mainly thinking that you could go and do your own initial review of the code here in github and leave comments for the reviewer drawing attention to changes like this.   If that's not something you want to do that's okay.